rest/config_legacy.go

```diff
@@ -93,17 +96,28 @@ type UnsupportedServerConfigLegacy struct {
 // ToStartupConfig returns the given LegacyServerConfig as a StartupConfig and a set of DBConfigs.
 // The returned configs do not contain any default values - only a direct mapping of legacy config options as they were given.
 func (lc *LegacyServerConfig) ToStartupConfig() (*StartupConfig, DbConfigMap, error) {
-
 	// find a database's credentials for bootstrap (this isn't the first database config entry due to map iteration)
 	bsc := &BootstrapConfig{}
 	for _, dbConfig := range lc.Databases {
 		if dbConfig.Server == nil || *dbConfig.Server == "" {
 			continue
 		}
+
+		server, username, password, err := legacyServerAddressUpgrade(*dbConfig.Server)
+		if err != nil {
+			return nil, DbConfigMap{}, err
```

Comment:
Callers are doing error checks anyway, but this will be a bit more defensive, to avoid callers thinking we successfully found zero databases.

```suggestion
			return nil, nil, err
```